pkg/syncer/metadatasyncer.go

```diff
@@ -459,7 +448,7 @@ func pvcUpdated(oldObj, newObj interface{}, metadataSyncer *metadataSyncInformer
 	}
 	migrationEnabled := metadataSyncer.coCommonInterface.IsFSSEnabled(ctx, common.CSIMigration)
 	// Verify if csi migration is ON and check if there is any label update or migrated-to annotation was received for the PVC
-	if migrationEnabled && pv.Spec.VsphereVolume != nil {
+	if pv.Spec.VsphereVolume != nil && migrationEnabled {
```

Comment:
Can you exchange the feature gate for this entire file?

Reply:
Done